docs/Spark/extensibility/gem-builder/gem-builder.md

```diff
@@ -320,14 +323,15 @@ Depending on what kind of Gem is being created, either a `Dialog` or a `DatasetD
 
 - The **Transformation Dialog**: The Dialog for Transformation Gems (any Gem that is not a Dataset Gem) is created using the `dialog` method, which must return a Dialog object.
 
-- The **Dataset Dialog**: The Dialog for a [Source/Target](../gems/source-target/) Gem is a `DatasetDialog` object. You will need to have `source` and `target` methods defined.
+- The **Dataset Dialog**: The Dialog for a [Source/Target](../../gems/source-target/source-target.md) Gem is a `DatasetDialog` object. You will need to have `source` and `target` methods defined.
 
 Column Selector: This is a special property that you should add if you want to select the columns from UI and then highlight the used columns using the `onChange` function.
 It is recommended to try out this dialogue code in Gem builder UI and see how each of these elements looks in UI.
 
 ### Validation
 
-The `validate` method performs validation checks so that in the case where there's any issue with any inputs provided for the user an Error can be displayed. In our example case if the Filter condition is empty. Similarly, you can add any validation on your properties.
+The `validate` method performs validation checks for issues with user-provided inputs and displays errors or warnings if
+any are detected. In our example case if the Filter condition is empty. Similarly, you can add any validation on your properties.
```

Comment:
this text doesnt' make any sense: 
> In our example case if the Filter condition is empty. Similarly, you can add any validation on your properties.

and this is not a very helpful example of validation. let's please fix this before closing the PR. may need help from a DE to get a better example of validation

Reply:
Thank you for catching this. I've rewritten it slightly.

This example existed before this PR so it's out of the original change scope. Do you mind if we track your request here instead? https://app.asana.com/0/1203457878958265/1208449446000978 I've reached out to folks already to see who can provide a better example.